src/algebra/group/defs.lean

```diff
@@ -388,6 +425,28 @@ class add_comm_monoid (M : Type u) extends add_monoid M, add_comm_semigroup M
 @[protect_proj, ancestor monoid comm_semigroup, to_additive]
 class comm_monoid (M : Type u) extends monoid M, comm_semigroup M
 
+@[ext, to_additive]
+lemma comm_monoid.ext {M : Type*} (m₁ m₂ : comm_monoid M) (h_mul : m₁.mul = m₂.mul) : m₁ = m₂ :=
+begin
+  unfreezingI {
+    cases m₁ with mul₁ _ one₁ one_mul₁ mul_one₁ npow₁ npow_zero₁ npow_succ₁,
+    cases m₂ with mul₂ _ one₂ one_mul₂ mul_one₂ npow₂ npow_zero₂ npow_succ₂ },
+  change mul₁ = mul₂ at h_mul,
+  subst h_mul,
+  have h_one : one₁ = one₂,
+  { rw ←one_mul₂ one₁,
+    exact mul_one₁ one₂ },
+  subst h_one,
+  have h_npow : npow₁ = npow₂,
+  { ext n,
+    induction n with d hd,
+    { rw [npow_zero₁, npow_zero₂] },
+    { rw [npow_succ₁, npow_succ₂, hd] } },
+  subst h_npow,
+end
```

Comment:
Here's how to avoid repeating yourself:
```suggestion
@[to_additive]
lemma comm_monoid.to_monoid_injective {M : Type u} :
  function.injective (@comm_monoid.to_monoid M) :=
begin
  rintros ⟨⟩ ⟨⟩ h,
  congr'; injection h,
end

@[ext, to_additive]
lemma comm_monoid.ext {M : Type*} ⦃m₁ m₂ : comm_monoid M⦄ (h_mul : m₁.mul = m₂.mul) : m₁ = m₂ :=
comm_monoid.to_monoid_injective $ monoid.ext h_mul
```

I think `⦃⦄` are better for `ext` lemmas since that matches the implicitness of `function.injective`

Reply:
Thank you! (Leaving this unresolved for now until I implement this throughout.)